.github/workflows/e2e-test.yml

```diff
@@ -117,18 +116,20 @@ jobs:
           echo "KLM_IMAGE_REPO=dev" >> $GITHUB_ENV
       - name: Deploy LM local testing kustomize
         run: |
-          if make local-deploy-with-watcher IMG=europe-docker.pkg.dev/kyma-project/$KLM_IMAGE_REPO/lifecycle-manager:$KLM_VERSION_TAG; then
-            echo "KLM deployed successfully"
-          else
-            echo "Deploy encountered some error, will retry after 1 minute"
-            sleep 60
+          maxRetry=5
+          for i in $(seq 1 $maxRetry)
+          do
             if make local-deploy-with-watcher IMG=europe-docker.pkg.dev/kyma-project/$KLM_IMAGE_REPO/lifecycle-manager:$KLM_VERSION_TAG; then
-                echo "KLM deployed successfully"
+              echo "KLM deployed successfully"
+              exit 0
```

Comment:
`exit 0` will end the pipeline here, or am I wrong? Since this is not the last step in the pipeline, we should not use `exit` here.

Please use `break` instead. See [here](https://www.cyberciti.biz/faq/bash-for-loop/#Conditional_exit_with_break)

Reply:
no, it ends this step only

Reply:
Can you show me a reference to your described behaviour? Since in official docs, if ends the actions. https://docs.github.com/en/actions/creating-actions/setting-exit-codes-for-actions

And anyway, `exit 1` ends the action fully, so I would consider anyway using break, so it is more clear that this does not ends the actions, and continues on the next step.